src/rust/inetstack/protocols/udp/peer.rs

```diff
@@ -128,17 +124,18 @@ impl<const N: usize> UdpPeer<N> {
         offload_checksum: bool,
         arp: ArpPeer<N>,
     ) -> Result<Self, Fail> {
-        let send_queue: SharedQueue<SharedQueueSlot<DemiBuffer>> =
-            SharedQueue::<SharedQueueSlot<DemiBuffer>>::new(SEND_QUEUE_MAX_SIZE);
-        let future = Self::background_sender(
-            rt.clone(),
+        let send_queue: SharedQueue<N> = SharedQueue::<N>::new(SEND_QUEUE_MAX_SIZE);
+        let mut rng: SmallRng = SmallRng::from_seed(rng_seed);
+        let ephemeral_ports: EphemeralPorts = EphemeralPorts::new(&mut rng);
```

Comment:
Any reason for moving these around?

Reply:
Nope, I just thought I would initialize everything before allocating the task as a rule, in case they need to be passed into the task.